### PR TITLE
Adds flexible type registry

### PR DESCRIPTION
**Pull request by jphenow**
Thoughts?

Proposing so the library doesn't have to worry about any other scalars than listed, but it'd be great if I didn't have to `ToString()` every time I have a Guid on the API.

Happy to touch up further to match project style and add tests, just wanted to get a rough implementation down before I forgot and moved on.

### CODE REVIEW

**Comment by joemcbride** [comment]
👍 I would probably change the `Stack` to a `Dictionary` + `TryGetValue` for lookup.

**Comment by jphenow** [comment]
The purpose of the Stack was simply so any added classes that overlap would default to user-added while retaining the original (in case at some point the system would want to force the system-default) but that's overkill now that I say it loud - especially since it isn't currently doing that.

Alright, yea, I'll switch it :-)

File: src/GraphQL/TypeExtensions.cs [hunk 1]
```diff
@@ -179,6 +143,8 @@ public static Type GetGraphTypeFromType(this Type type, bool isNullable = false)
                 graphType = listType.MakeGenericType(elementType);
             }
 
+            graphType = GraphQL.GraphTypeRegistry.Get(type);
```
Comment (joemcbride) [inline src/GraphQL/TypeExtensions.cs]: I believe this needs to be moved above the `type.IsArray` check to fix the tests.

**Comment by jphenow** [comment]
Woot! Thanks!

**Pull request merged**